include/yaramod/types/regexp.h

```diff
@@ -437,7 +548,12 @@ class RegexpGroup : public RegexpUnit
 class RegexpConcat : public RegexpUnit
 {
 public:
-	RegexpConcat(std::vector<std::shared_ptr<RegexpUnit>>&& units) : _units(std::move(units)) {}
+	RegexpConcat(std::vector<std::shared_ptr<RegexpUnit>>&& units)
+		: _units(std::move(units))
+	{
+		for (auto unit : _units)
```

Comment:
Do `for (const auto& unit : _units)` or `for (auto&& unit : _units)`. You are making copies of shared pointers for each iteration right now.